sdk/src/main/java/com/podio/sdk/domain/field/ReminderRecurrenceField.java

```diff
@@ -0,0 +1,167 @@
+        private final Integer step;
+        private final List<String> days;
+        private final String repeat_on;
+        private final String name;
+        private final String until;
+
+        public Value(Integer remind_delta, Integer step, List<String> days, String repeat_on, String name, String until) {
+            this.remind_delta = remind_delta;
+            this.step = step;
+            this.days = days;
+            this.until = until;
+            this.name = name;
+            this.repeat_on = repeat_on;
+        }
+
+        @Override
+        public Map<String, Object> getCreateData() {
+            //Is not used
+            return null;
+        }
+
+        public Map<String, Object> getReminderData() {
+            if (remind_delta == null || remind_delta == -1) {
+                return null;
+            }
+
+            Map<String, Object> reminderData = new HashMap<>();
+            reminderData.put(REMIND_DELTA, remind_delta);
+
+            return reminderData;
+        }
+
+        public Map<String, Object> getRecurrenceData() {
+            if(name == null) {
+                return null;
+            }
+
+            Map<String, Object> recurrenceData = new HashMap<>();
+            Map<String, Object> configData = new HashMap<>();
+
+            recurrenceData.put(NAME, name);
+            if(name.equalsIgnoreCase(WEEKLY)) {
+                configData.put(DAYS, days);
+                recurrenceData.put(CONFIG, configData);
+                recurrenceData.put(STEP, step);
+            }
+            else if(name.equalsIgnoreCase(MONTHLY)) {
+                configData.put(REPEAT_ON, repeat_on);
+                recurrenceData.put(CONFIG, configData);
+                recurrenceData.put(STEP, step);
+            }
+            recurrenceData.put(UNTIL, until);
+
+            return recurrenceData;
+        }
+
+        public Integer getRemindDelta() {
+            return remind_delta;
+        }
+
+        public Integer getStep() {
+            return step;
+        }
+
+        public List<String> getDays() {
+            return days;
+        }
+
+        public String getName() {
+            return name;
+        }
+
+        public String getUntil() {
+            return until;
+        }
+
+        public String getRepeatOn() {
+            return repeat_on;
+        }
+    }
+
+    private final ArrayList<ReminderRecurrenceField.Value> values;
+
+    public ReminderRecurrenceField() {
+        super(NAME);
+        this.values = new ArrayList<>();
+    }
+
+    @Override
+    public void addValue(Value value) {
+        if (values != null && !values.contains(value)) {
+            values.clear();
+            values.add(value);
+        }
+    }
+
+    @Override
+    public void setValues(List<Value> values) {
+        this.values.clear();
+        this.values.addAll(values);
+    }
+
+    @Override
+    public Value getValue(int index) {
+        return values != null ? values.get(index) : null;
```

Comment:
Well, we make sure to set values as soon as we create the values, but I guess it is safe to throw in a index check 👍